hack/charts/cluster-api-operator/templates/infra-conditions.yaml

```diff
@@ -49,6 +49,15 @@ metadata:
     "helm.sh/hook-weight": "2"
 {{- with .Values.configSecret }}
 spec:
+{{- if $.Values.manager }}
+  manager:
+{{- if $.Values.manager.featureGates.kubeadm }}
```

Comment:
can we make `kubeadm` configurable here? for users that are not using kubeadm for bootstrapping

Reply:
@alexander-demicev 
Sorry, I don't seem to understand correctly what you mean.
Could you please explain in more detail?
Thx.

Reply:
As changes may be made, the test is written after hearing the explanation.
Thx.

Reply:
@alexander-demicev remind 🙇‍♂️

Reply:
@kahirokunn current implementation always assumes that `manager.featureGates` dictionary will use `kubeadm` as a bootstrap provider and nothing else. However, there is a possibilty that, other bootstrap provider, like `rke2` could be used instead of `kubeadm`, so making   `manager.featureGates` configurable would makes sense here only if we could make the whole `ControlPlaneProvider` type (starting from line 42) configurable including` metadata.name` & `metadata.namespace`. 

@alexander-demicev perhaps we could keep it as is, since it is already kubeadm specific, WDYT?

Reply:
I wrote this because the file is specific to kubeadm
 I wrote this because it is a kubeadm-specific file, and I thought it would be confusing if I wrote it in a standard way.